app/controllers/concerns/check_query_count.rb

```diff
@@ -0,0 +1,48 @@
+module CheckQueryCount
+  class ExcessQueryCountException < StandardError; end
+
+  class CountChecker
+    MAX_QUERY_THRESHOLD = Rails.env.test? ? 30 : 300
```

Comment:
in this case, it might be better to use the 
/config/environments/development.rb (production.rb and test.rb) to set that value:

```
config.max_query_threshold = 30 # in test.rb

config.max_query_threshold = 300  # in development.rb, and production.rb
```

Reply:
Yeh - good shout - may make it configurable via an env var instead though